core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -2589,6 +2590,27 @@ public OptionalInt getMaxWriterTasks(Session session, String catalogName)
         return catalogMetadata.getMetadata(session).getMaxWriterTasks(session.toConnectorSession(catalogHandle));
     }
 
+    @Override
+    public WriterScalingOptions getNewTableWriterScalingOptions(Session session, QualifiedObjectName tableName, Map<String, Object> tableProperties)
+    {
+        Optional<CatalogMetadata> catalog = getOptionalCatalogMetadata(session, tableName.getCatalogName());
```

Comment:
You should use `io.trino.metadata.MetadataManager#getCatalogMetadataForWrite(io.trino.Session, java.lang.String)` instead (see `io.trino.metadata.MetadataManager#getNewTableLayout`)